source/elements/oneDPL/source/parallel_api/execution_policies.rst

```diff
@@ -180,5 +178,54 @@ as the template argument, otherwise unspecified.
 Return a policy object constructed from ``policy``, with a new kernel name provided as the template
 argument. If no policy object is provided, the new policy is constructed from ``dpcpp_default``.
 
+.. _exec-policy-type-trait:
+
+Execution Policy Type Trait
++++++++++++++++++++++++++++
+
+oneDPL provides type trait utilities to detect its execution policy types at compile time for the purpose of
+function overload resolution:
+
+.. code:: cpp
+
+  // Defined in <oneapi/dpl/execution>
+
+  namespace oneapi {
+    namespace dpl {
+
+      template <class T>
+      struct is_execution_policy { /*see below*/ };
+
+      template <class T>
+      constexpr bool is_execution_policy_v = oneapi::dpl::is_execution_policy<T>::value;
+
+      namespace execution {
+
+        template <class T>
+        struct is_execution_policy { /*see below*/ };
+
+        template <class T>
+        constexpr bool is_execution_policy_v = oneapi::dpl::execution::is_execution_policy<T>::value;
+
+      }
+    }
+  }
+
+``oneapi::dpl::is_execution_policy`` and ``oneapi::dpl::execution::is_execution_policy`` must be treated as name aliases
+to the same class template. It is unspecified in which namespace the underlying class template and its specializations
+are defined.
+
+.. note::
+   The ``oneapi::dpl::execution::is_execution_policy`` class originated in the oneDPL specification version 1.0,
+   while ``oneapi::dpl::is_execution_policy`` has been added later to better align with the C++ standard.
+   
+   For writing new code, use of the type trait utilities in ``namespace oneapi::dpl`` is strongly recommended. Those
+   in ``namespace oneapi::dpl::execution`` are provided for backward compatibility and may be deprecated in the future.
+
+``is_execution_policy<T>`` must have the characteristics of ``std::true_type`` if ``T`` is one of the above specified or
+implementation-defined oneDPL execution policy types, otherwise it must have the characteristics of ``std::false_type``.
```

Comment:
What's the point of this super generic and kind of vague wording?

I specifically mean "_is one of the above specified or implementation-defined oneDPL execution policy types_".

Reply:
Compare with the C++ standard wording:
> `is_execution_policy<T>` is a Cpp17UnaryTypeTrait with a base characteristic of `true_type` if `T` is the type of a standard or implementation-defined execution policy, otherwise `false_type`.

It is assumed that an implementation can define additional policy types, for which the trait also resolves to `true_type`.

Reply:
I see. Given that this is a specification, we allow the implementations of oneDPL spec to have their own _implementation-defined_ execution policies, if they will.

Thanks.